src/unlimited.cpp

```diff
@@ -1053,21 +1054,23 @@ UniValue settrafficshaping(const UniValue& params, bool fHelp)
 // This way we avoid having to lock cs_main so often which tends to be a bottleneck.
 void IsChainNearlySyncdInit() 
 {
-    LOCK2(cs_main, cs_ischainnearlysyncd);
-    if (!pindexBestHeader) fIsChainNearlySyncd = false;  // Not nearly synced if we don't have any blocks!
-    else
-      {
-      if(chainActive.Height() < pindexBestHeader->nHeight - 2)
-        fIsChainNearlySyncd = false;
-      else
-        fIsChainNearlySyncd = true;
-      }
+    bool flag = fIsChainNearlySyncd.load();
+    LOCK(cs_main);
+    if (!pindexBestHeader) {
+        // Not nearly synced if we don't have any blocks!
+        while (flag && !fIsChainNearlySyncd.compare_exchange_weak(flag, false));
```

Comment:
I think this code would work, but AFAIK the purpose of compare_exchange_xxx is handle a situation where the code needs to use the current value of the atomic.  For example for a read, add, write, you'd do:
int current = val.load();
while (!fIsChainNearlySyncd.compare_exchange_weak(val, val+5));

But if you just need to set the value of the object, irrespective of its current value, then use the "store" API: http://en.cppreference.com/w/cpp/atomic/atomic/store

So in a nutshell, I think it would be a lot more clear to replace lines 1061, 1065 with
fIsChainNearlySyncd.store(false)
and 1067 with .store(true)